include/mxnet/c_api.h

```diff
@@ -8,9 +8,15 @@
 
 #ifdef __cplusplus
 #define MXNET_EXTERN_C extern "C"
```

Comment:
MXNET_EXTERN_C approach is not a standard one. please take the chance to fix it. This also reduces the current patch to very limit lines of changes.

More standard way to wrap extern "C" is like this: (take from png.h)

/* Inhibit C++ name-mangling for libpng functions but not for system calls. */
#ifdef __cplusplus
extern "C" {
#endif /* __cplusplus */